Crash importing vault_jwt_auth_backend_role if bound_audiences is nil

### Terraform Version
```
Terraform v0.12.12
+ provider.vault v2.5.0
```

### Affected Resource(s)
- vault_jwt_auth_backend_role

### Terraform Configuration Files
```hcl
resource "vault_jwt_auth_backend_role" "azure_default_role" {
  backend         = "oidc"
  bound_audiences = []
  role_name       = "default"
  user_claim      = "email"
  groups_claim    = "groups"
  oidc_scopes = [
    "https://graph.microsoft.com/.default",
    "email"
  ]
  allowed_redirect_uris = [
    "${local.vault_addr}/ui/vault/auth/oidc/oidc/callback",
    "http://localhost:8250/oidc/callback"
  ]

  token_policies = []
  token_ttl      = 3600
  token_max_ttl  = 86400
}
```

### Debug Output
None

### Panic Output
```
vault_jwt_auth_backend_role.azuread_default_role: Importing from ID "auth/oidc/role/default"...
vault_jwt_auth_backend_role.azuread_default_role: Import prepared!
  Prepared vault_jwt_auth_backend_role for import
vault_jwt_auth_backend_role.azuread_default_role: Refreshing state... [id=auth/oidc/role/default]

Error: rpc error: code = Unavailable desc = transport is closing


panic: interface conversion: interface {} is nil, not []interface {}
2019-11-04T09:19:38.425-0600 [DEBUG] plugin.terraform-provider-vault_v2.5.0_x4:
2019-11-04T09:19:38.425-0600 [DEBUG] plugin.terraform-provider-vault_v2.5.0_x4: goroutine 82 [running]:
2019-11-04T09:19:38.425-0600 [DEBUG] plugin.terraform-provider-vault_v2.5.0_x4: github.com/terraform-providers/terraform-provider-vault/vault.jwtAuthBackendRoleRead(0xc0002a4000, 0x1ec5aa0, 0xc0002e05b0, 0xc0002a4000, 0x0)
2019-11-04T09:19:38.425-0600 [DEBUG] plugin.terraform-provider-vault_v2.5.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vault/resource_jwt_auth_backend_role.go:312 +0x1ea6
```

https://gist.github.com/sean-nixon/82da78bc02dcd99bf939e28808f48e06

### Expected Behavior
Successfully import resource

### Actual Behavior
Crash

### Steps to Reproduce
Create a role using the vault CLI with no bound_audiences set as is specified in the configuration guide for OIDC (https://www.vaultproject.io/docs/auth/jwt.html#oidc-configuration-troubleshooting)

```
> vault write "auth/oidc/role/default" \
    user_claim="email" \
    groups_claim="groups" \
    oidc_scopes="https://graph.microsoft.com/.default,email" \
    allowed_redirect_uris="$VAULT_ADDR/ui/vault/auth/oidc/oidc/callback" \
    allowed_redirect_uris="http://localhost:8250/oidc/callback" \
    ttl=1h

> vault read auth/oidc/role/default
Key                        Value
---                        -----
allowed_redirect_uris      [<VAULT_ADDR>/ui/vault/auth/oidc/oidc/callback http://localhost:8250/oidc/callback]
bound_audiences            [c78a2df8-3bd2-4954-af76-47d1c3c7b777]
bound_claims               <nil>
bound_subject              n/a
claim_mappings             <nil>
clock_skew_leeway          0
expiration_leeway          0
groups_claim               groups
not_before_leeway          0
oidc_scopes                [https://graph.microsoft.com/.default email]
role_type                  oidc
token_bound_cidrs          []
token_explicit_max_ttl     0s
token_max_ttl              0s
token_no_default_policy    false
token_num_uses             0
token_period               0s
token_policies             []
token_ttl                  1h
token_type                 default
ttl                        1h
user_claim                 email
verbose_oidc_logging       false
```

Try to import the resource.

```
terraform import vault_jwt_auth_backend_role.azuread_default_role "auth/oidc/role/default"
```

Updating the role to set a value for `bound_audiences` fixes the crash.

--- a/vault/resource_jwt_auth_backend_role.go
+++ b/vault/resource_jwt_auth_backend_role.go
@@ -333,10 +333,12 @@ func jwtAuthBackendRoleRead(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
-	boundAuds := util.JsonStringArrayToStringArray(resp.Data["bound_audiences"].([]interface{}))
-	err = d.Set("bound_audiences", boundAuds)
-	if err != nil {
-		return fmt.Errorf("error setting bound_audiences in state: %s", err)
+	if resp.Data["bound_audiences"] != nil {
+		boundAuds := util.JsonStringArrayToStringArray(resp.Data["bound_audiences"].([]interface{}))
+		err = d.Set("bound_audiences", boundAuds)
+		if err != nil {
+			return fmt.Errorf("error setting bound_audiences in state: %s", err)
+		}
 	}
 
 	d.Set("user_claim", resp.Data["user_claim"].(string))

--- a/vault/resource_jwt_auth_backend_role.go
+++ b/vault/resource_jwt_auth_backend_role.go
@@ -339,6 +339,8 @@ func jwtAuthBackendRoleRead(d *schema.ResourceData, meta interface{}) error {
 		if err != nil {
 			return fmt.Errorf("error setting bound_audiences in state: %s", err)
 		}
+	} else {
+		d.Set("bound_audiences", make([]string, 0))
 	}
 
 	d.Set("user_claim", resp.Data["user_claim"].(string))

--- a/vault/resource_jwt_auth_backend_role_test.go
+++ b/vault/resource_jwt_auth_backend_role_test.go
@@ -288,10 +288,6 @@ func TestAccJWTAuthBackendRoleOIDC_full(t *testing.T) {
 					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
 						"bound_subject", "sl29dlldsfj3uECzsU3Sbmh0F29Fios1@client"),
 					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
-						"bound_audiences.#", "1"),
-					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
-						"bound_audiences.2478800941", "https://myco.test"),
-					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
 						"user_claim", "https://vault/user"),
 					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
 						"groups_claim", "https://vault/groups"),
@@ -610,7 +606,7 @@ resource "vault_jwt_auth_backend" "jwt" {
   oidc_client_secret = "secret"
   lifecycle {
   ignore_changes = [
-     # Ignore changes to odic_clie_secret inside the tests
+     # Ignore changes to oidc_client_secret inside the tests
      "oidc_client_secret"
     ]
   }
@@ -624,7 +620,6 @@ resource "vault_jwt_auth_backend_role" "role" {
 
   bound_subject = "sl29dlldsfj3uECzsU3Sbmh0F29Fios1@client"
   token_bound_cidrs = ["10.148.0.0/20", "10.150.0.0/20"]
-  bound_audiences = ["https://myco.test"]
   user_claim = "https://vault/user"
   groups_claim = "https://vault/groups"
   token_policies = ["default", "dev", "prod"]

--- a/website/docs/r/jwt_auth_backend_role.html.md
+++ b/website/docs/r/jwt_auth_backend_role.html.md
@@ -45,7 +45,6 @@ resource "vault_jwt_auth_backend_role" "example" {
   role_name       = "test-role"
   token_policies  = ["default", "dev", "prod"]
 
-  bound_audiences       = ["https://myco.test"]
   user_claim            = "https://vault/user"
   role_type             = "oidc"
   allowed_redirect_uris = ["http://localhost:8200/ui/vault/auth/oidc/oidc/callback"]
@@ -60,8 +59,8 @@ The following arguments are supported:
 
 * `role_type` - (Optional) Type of role, either "oidc" (default) or "jwt".
 
-* `bound_audiences` - (Required) List of `aud` claims to match
-  against. Any match is sufficient.
+* `bound_audiences` - (Required for roles of type `jwt`, optional for roles of
+  type `oidc`) List of `aud` claims to match against. Any match is sufficient.
 
 * `user_claim` - (Required) The claim to use to uniquely identify
   the user; this will be used as the name for the Identity entity alias created

